BasicPlaybackSampleApp/BasicPlaybackSampleApp/Players/BasicSimplePlayerViewController.m

```diff
@@ -96,8 +97,34 @@ - (void)viewDidLoad {
   [self addPlayerViewController:self.ooyalaPlayerViewController];
   
   // Load the video
-  [self.ooyalaPlayerViewController.player setEmbedCode:self.embedCode];
-  [self.ooyalaPlayerViewController.player play];  
+  //[self.ooyalaPlayerViewController.player setEmbedCode:self.embedCode];
+  //Deprecated API. Remove this call when SDK version become more then 4.46.0_GA. Uncomment code with Asynchronous method instead.
+  //[self.ooyalaPlayerViewController.player play];
+  
+  //new API. Uncomment when SDK version become more then 4.46.0_GA
+  __weak typeof(self) weakSelf = self;
+  
+  [self.ooyalaPlayerViewController.player setEmbedCode:self.embedCode withCallback:^(OOOoyalaError *error) {
+    LOG(@"✅ got callback. embed: %@, is success: %@. But it doesn't mean that status is 'AVPlayerItemStatusReadyToPlay'", weakSelf.ooyalaPlayerViewController.player.currentItem.embedCode, (error == nil) ? @"YES" : @"NO");
+    if (error) {
+      LOG(@"❌ error: %@", error.debugDescription);
+    }
+  }];
+  
+  //OS: to avoid calling '[weakSelf.player play]' until player really ready to play
+  void (^expectedBlock) (OOVideo *currentItem);
+  expectedBlock = ^ (OOVideo *currentItem) {
+    LOG(@"✅ got expectedBlock");
+    if ([currentItem.embedCode isEqualToString:weakSelf.embedCode]) {
+      NSLog(@"✅ aseet with embed code %@", currentItem.embedCode);
+      [weakSelf.ooyalaPlayerViewController.player play];
+      //OS: block must be removed after '[weakSelf.player play]', to prevent ignition from OOBaseStreamPlayer's KVO 'AVPlayerItemStatusReadyToPlay'
+      weakSelf.ooyalaPlayerViewController.player.currentItemChangedCallback = nil;
+    } else {
+      NSLog(@"❌ asset with embed code %@ must be in hospital and take injections", currentItem.embedCode);
```

Comment:
What?

Reply:
Sorry, it's copy-paste from my sandbox's project that manages different types of blocks : )